### PR TITLE
inject: add `config.linkerd.io/shutdown-grace-period` annotation

### PR DESCRIPTION
**Pull request by hawkw**
PR linkerd/linkerd2-proxy#1815 added support for a
`LINKERD2_PROXY_SHUTDOWN_GRACE_PERIOD` environment variable that
configures the proxy's maximum grace period for graceful shutdown. This
is intended to ensure that if a proxy is shut down, it will eventually
terminate in a relatively timely manner, even if some stubborn
connections don't close gracefully.

This branch adds support for a `config.linkerd.io/shutdown-grace-period`
annotation that can be used to override the default grace period
duration.

Hopefully I've added this everywhere it needs to be added --- please let
me know if I've missed anything!

### CODE REVIEW

File: cli/cmd/testdata/install_controlplane_tracing_output.golden [hunk 1]
```diff
@@ -735,7 +736,7 @@ spec:
     spec:
       nodeSelector:
         kubernetes.io/os: linux
-
```
Comment (adleong) [inline cli/cmd/testdata/install_controlplane_tracing_output.golden:738]: What caused this unrelated whitespace change, I wonder?  Weird........

**Review by adleong — COMMENTED** [review]
I think we also want to add this to charts/linkerd-control-plane/values.yaml so that this shows up in the chart readme as a settable install value.

**Comment by hawkw** [comment]
> I think we also want to add this to charts/linkerd-control-plane/values.yaml so that this shows up in the chart readme as a settable install value.

Ah, thanks, I knew I was missing something!

File: cli/cmd/testdata/install_controlplane_tracing_output.golden (hunk 1 above)
Comment (hawkw) [reply cli/cmd/testdata/install_controlplane_tracing_output.golden:738]: ugh, i think i might have a VS Code plugin that strips whitespace from blank lines, sorry. i can try to back that out.

**Review by adleong — APPROVED** [review]

**Comment by hawkw** [comment]
Okay, I've backed out the accidental whitespace change and updated `charts/linkerd-control-plane/values.yaml` and the chart README. I think this should be all good now.

I'm assuming we don't want to add a CLI flag for overriding this and can expect people to set it through the annotation?

**Comment by hawkw** [comment]
Integration test failure is almost certainly unrelated, [looks like getting `version.json` timed out](https://github.com/linkerd/linkerd2/runs/7416814782?check_suite_focus=true#step:8:318).

For some reason, I don't seem to have the option to restart that job, is that disabled in this repo or something?

**Review by olix0r — APPROVED** [review]
🎉
